### PR TITLE
video-test: More robust test cleanup code

### PR DESCRIPTION
**Pull request by aghassemi**
(possibly) fixes https://github.com/ampproject/amphtml/issues/9909, https://github.com/ampproject/amphtml/issues/9910 and https://github.com/ampproject/amphtml/issues/9904

### CODE REVIEW

File: test/integration/test-video-players-helper.js [hunk 1]
```diff
@@ -256,12 +256,16 @@ export function runVideoPlayerIntegrationTests(
   }
 
   function cleanUp() {
```
Comment (jridgewell) [inline test/integration/test-video-players-helper.js]: None of this should be necessary. Any iframes appended to the test runner will be cleaned up automatically.

Comment (aghassemi) [inline test/integration/test-video-players-helper.js]: looked into it. It doesn't cleanup iframes in Safari for some reason..
from `_init_tests.js_`
```
  if (!platformFor(window).isSafari()) {
    cleanupTagNames.push('iframe');
  }
```

let me remove this and enable cleanup it for Safari and see how it goes.

Comment (jridgewell) [inline test/integration/test-video-players-helper.js]: That's because any timeout in the iframe will still fire, causing all kinds of mayhem.

Comment (aghassemi) [inline test/integration/test-video-players-helper.js]: Problem with this particular test is that iOS can't play more than one video at a time, so I really need to remove the video element from the previous test's iframe if iframe itself is not being removed. Anyway just removed `if (!platformFor(window).isSafari()) {` part, let's see what happens with the tests. If there are issues, then we should go with the first version of this PR and keep the cleanup method.

**Comment by aghassemi** [comment]
@jridgewell @zhouyx lots of unrelated integration tests failing, I have run this 4 times with no luck. force merge?

**Comment by zhouyx** [comment]
😭 😭 😭  let's see after #9930 gets merged

**Comment by zhouyx** [comment]
@aghassemi I'll monitor this. But I think you still get change request from @jridgewell ?

**Comment by jridgewell** [comment]
You'll need to rebase.

**Pull request merged**